routes/api/contacts.js

```diff
@@ -1,25 +1,110 @@
-const express = require('express')
+const express = require("express");
+const Joi = require("joi");
+const { nanoid } = require("nanoid/non-secure");
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const schemaPost = Joi.object({
+  name: Joi.string().alphanum().min(3).max(30).required(),
+  phone: Joi.string()
+    .pattern(/^(?:\+38)?0\d{9}$/)
+    .required()
+    .messages({
+      "string.pattern.base": "Invalid phone number format",
+      "any.required": "Phone number is required",
+    }),
+  email: Joi.string()
+    .email({
+      minDomainSegments: 2,
+      tlds: { allow: ["com", "net"] },
+    })
+    .required(),
+});
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const schemaPut = Joi.object({
+  name: Joi.string().alphanum().min(3).max(30),
+  phone: Joi.string()
+    .pattern(/^(?:\+38)?0\d{9}$/)
+    .messages({
+      "string.pattern.base": "Invalid phone number format",
+      "any.required": "Phone number is required",
+    }),
+  email: Joi.string().email({
+    minDomainSegments: 2,
+    tlds: { allow: ["com", "net"] },
+  }),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const router = express.Router();
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  const contacts = await listContacts();
+  res.status(200).json(contacts);
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const contact = await getContactById(req.params.contactId);
+    res.status(200).json(contact);
```

Comment:
this trycatch might be stored in controller function. Router should only invoke function. But not define another one